Sources/SourceKitLSP/SourceKitServer.swift

```diff
@@ -1172,6 +1177,24 @@ extension SourceKitServer {
     languageService.inlayHint(req)
   }
 
+  /// Converts a location from the symbol index to an LSP location.
+  /// 
+  /// - Parameter location: The symbol index location
+  /// - Returns: The LSP location
+  private func indexToLSPLocation(_ location: SymbolLocation) -> Location? {
+    guard !location.path.isEmpty else { return nil }
+    return Location(
+      uri: DocumentURI(URL(fileURLWithPath: location.path)),
+      range: Range(Position(
+        // 1-based -> 0-based
+        // Note that we still use max(0, ...) as a fallback if the location is zero.
+        line: max(0, location.line - 1),
+        // FIXME: we need to convert the utf8/utf16 column, which may require reading the file!
+        utf16index: max(0, location.utf8Column - 1)
```

Comment:
This change btw fixes a bug with definition lookup too, e.g. ctrl+clicking a standard library function like `print` would (at least on my Mac) yield `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/lib/swift/Swift.swiftmodule/arm64e-apple-macos.swiftinterface` along with `(0, 0)` as a position (which would subsequently get turned into `(-1, -1)` by this conversion and cause errors on the LSP client-side.

Reply:
Aside from that it would of course be cool to have proper positions even within a `*.swiftinterface`, but for now having a fallback at least lets us link to the file correctly.

Reply:
Yes, having proper locations in`.swiftinterface` files would be great, but IIRC that’s a bigger undertaking. For now, I think this is a good fix.